plugins/read_roots/root_loader_main.ml

```diff
@@ -1,24 +1,33 @@
-module File = struct
-  open Option.Monad_infix
+open Core_kernel.Std
+open Bap.Std
+open Bap_future.Std
 
-  let split_right ~after str =
-    match String.rindex str after with
-    | None -> str,None
-    | Some n -> String.(subo ~len:n str, Some (subo ~pos:(n+1) str))
+include Self ()
 
-  let extension s = split_right ~after:'.' s |> snd
+let extract filename arch =
+  let width = Arch.addr_size arch |> Size.in_bits in
+  In_channel.with_file filename ~f:(fun ch ->
+      Sexp.input_sexps ch |> List.map ~f:Int64.t_of_sexp |>
+      List.map ~f:(Addr.of_int64 ~width) |>
+      Seq.of_list)
 
-  (* input file format is [<name>.<fmt>[-<ver>]]. The `<fmt>` part is
-     mandatory and should match with some reader, registered for Addr,
-     data type. The `<ver>` part is optional, and defines format
-     version. Example: "functions.sexp" or "functions.sexp-1.0".
-     The example above will use sexp serializer to read data from file.
-  *)
-  let rooter filename =
-    let name,ver = split_right ~after:'-' filename in
-    extension name >>= fun fmt -> Addr.find_reader fmt >>| fun _ ->
-    In_channel.with_file filename ~f:(fun chan ->
-        Addr.Io.load_all ?ver ~fmt chan |>
-        Seq.of_list |>
-        create)
-end
+let register filename =
+  Stream.map Project.Info.arch (fun arch ->
+      Ok (Rooter.create (extract filename arch))) |>
+  Rooter.Factory.register "file"
```

Comment:
the rooter name shall be consistent with the the read_symbols plugin, i.e., it should be `file:<filename>`